Siren Pectra Feature Updates (#7396)
sigp/lighthouse: book/src/ui_usage.md

![exit](imgs/ui-val-exit.png)

### Deposit and Import new Validators

Siren's deposit flow aims to create a smooth and easy process for depositing and importing a new Lighthouse validator. The process is seperated into 6 main steps:

#### Validator Setup

- First, select the number of validators you wish to create, ensuring you connect a wallet with sufficient funds to cover each validator deposit. For each validator candidate, you can set a custom name and optionally enable the `0x02` withdrawal credential flag, which indicates to the deposit contract that the validator will compound and have an increased `MAX_EFFECTIVE_BALANCE`.

![deposit-step-1](imgs/ui-dep-1.png)

#### Phrase Verification

- Enter a valid mnemonic phrase to generate corresponding deposit JSON and keystore objects. This is a sensitive step; copying and pasting your mnemonic phrase is not recommended. This information is never stored or transmitted through any communication channel.

![deposit-step-2](imgs/ui-dep-2.png)

#### Mnemonic Indexing

> - The mnemonic index is as important as the mnemonic phrase; reusing existing or exited indices may result in invalid deposits. Each index combined with the mnemonic phrase generates a deterministic public key, which Siren validates by checking against the Beacon Node. Since newly submitted deposits may not immediately appear on the Beacon Node, Siren provides BeaconCha links for secondary confirmation.

Reviewer: Should we use the actual url here instead?

```suggestion
- The mnemonic index is as important as the mnemonic phrase; reusing existing or exited indices may result in invalid deposits. Each index combined with the mnemonic phrase generates a deterministic public key, which Siren validates by checking against the Beacon Node. Since newly submitted deposits may not immediately appear on the Beacon Node, Siren provides beaconcha.in links for secondary confirmation.
```

> - The mnemonic index is as important as the mnemonic phrase; reusing existing or exited indices may result in invalid deposits. Each index combined with the mnemonic phrase generates a deterministic public key, which Siren validates by checking against the Beacon Node. Since newly submitted deposits may not immediately appear on the Beacon Node, Siren provides [Beaconcha.in](https://beaconcha.in) links for secondary confirmation.

Reviewer 2: How does reusing existing index cause invalid deposit? IIRC the deposit will still go through, only to be withdrawn via validator sweep if the validator has 0x01 withdrawal credentials.

Reviewer 3: Yep you're right CK. Invalid might not be the right word, maybe `incorrect`. 

For example, the withdrawal address will not be upheld, instead it will use the original one from the 0x01 creds.

Reviewer 2: Not sure about "incorrect deposits", I was thinking "redundant deposits"

> - The mnemonic index is as important as the mnemonic phrase; reusing existing or previously exited indices directs deposits to existing validators and requires additional steps to recover those funds. Each index combined with the mnemonic phrase generates a deterministic public key, which Siren validates by checking against the Beacon Node. Since newly submitted deposits may not immediately appear on the Beacon Node, Siren provides [Beaconcha.in](https://beaconcha.in) links for secondary confirmation.

Reviewer 2: Just add a "may" here. Because if reuse existing indices with 0x01 withdrawal credentials, there is nothing to be done, the deposit will just be swept automatically. But with 0x00 then there is "the additional steps" here, i.e., BLS To Execution Change

```suggestion
- The mnemonic index is as important as the mnemonic phrase; reusing existing or previously exited indices directs deposits to existing validators and may requires additional steps to recover those funds. Each index combined with the mnemonic phrase generates a deterministic public key, which Siren validates by checking against the Beacon Node. Since newly submitted deposits may not immediately appear on the Beacon Node, Siren provides [Beaconcha.in](https://beaconcha.in) links for secondary confirmation.
```

Reviewer 2: It should be `may require`, feel free to change it

Author: thats good too

![deposit-step-3](imgs/ui-dep-3.png)

#### Withdrawal Credentials

> - Next, set the withdrawal and suggested fee recipient addresses. In the basic view, you can conveniently set both values to the same address, or switch to the advanced view to specify them separately. You may apply these settings uniformly to all validators or individually per candidate. Each value can be verified by connecting the relevant wallet and signing a valid message. Skipping verification is not recommended, as these credentials will receive all validator funds and cannot be changed later.

Reviewer 2: A small suggestion to make it more specific to mention that withdrawal address cannot be changed, since fee recipient address can be changed

```suggestion
- Next, set the withdrawal and suggested fee recipient addresses. In the basic view, you can conveniently set both values to the same address, or switch to the advanced view to specify them separately. You may apply these settings uniformly to all validators or individually per candidate. Each value can be verified by connecting the relevant wallet and signing a valid message. Skipping verification is not recommended, as the withdrawal address will receive the staked validator funds and cannot be changed later.
```

![deposit-step-5](imgs/ui-dep-5.png)

#### Sign and Deposit

- Finally, complete each deposit by connecting a wallet with sufficient funds to Siren and signing the transaction. Upon successful inclusion of the deposit in the next block, Siren automatically imports the validator using the provided keystore credentials. Once imported, your validator will appear in Siren when the Beacon Node processes the transaction and enters the deposit queue. Processing time may vary depending on the queue length, potentially taking several days. Siren maintains a record of the deposit transaction for your review during this period.

![deposit-step-6](imgs/ui-dep-6.png)

### Consolidate Validator

`EIP-7251` increases the `MAX_EFFECTIVE_BALANCE` limit up to `2048 ETH`, allowing validators with `0x02` withdrawal credentials to consolidate funds from multiple exited validators. Siren facilitates requests to a consolidation contract, enabling validators to upgrade their withdrawal credentials and merge several validators into one compounding target validator.

![consolidation-target](imgs/consolidation-target.png)

#### Eligibility requirements for consolidation

> - Validators must have at least `0x01` withdrawal credentials. Validators with `0x00` credentials must first perform a BLS Execution Change.

Reviewer: Can we link to some docs on performing a BLS Execution Change here?

- Target validators with `0x01` withdrawal credentials must initiate a self-consolidation request to upgrade credentials to `0x02`, enabling them to accept funds and benefit from the increased balance cap.

> - Validators selected for merging must already be eligible for withdrawal.

Reviewer 2: This is not very clear. Do you mean the source validator here? 

Does 'be eligible for withdrawal' means having 0x01 withdrawal credentials? If so, that's the same thing as the first point above

Author: in the spec it has `if current_epoch < source_validator.activation_epoch + SHARD_COMMITTEE_PERIOD`

my understanding was these two combined equaled the exit epoch of a validator

Reviewer 2: oh that's what you mean. Yes, the validator must be active long enough (~ 1 day) to be eligible to exit. Perhaps change the wording to mentioning "long enough to be eligible to exit" will be clearer?

![consolidation-source](imgs/consolidation-source.png)

#### Post-consolidation

> - Consolidated validators will exit automatically, and their funds will be transferred to the target validator.

Reviewer 2: ```suggestion
- The source validator in the consolidation will exit automatically, and their funds will be transferred to the target validator.
```

- Validators consolidated under the new credentials (`0x02`) will no longer participate in automatic partial withdrawal sweeps. Instead, withdrawal requests must be explicitly submitted to the withdrawal contract as defined in `EIP-7002`.

### Partial Validator Withdrawal

> `EIP-7002` enables partial withdrawals from validators with `0x02` withdrawal credentials and balances exceeding the `MIN_ACTIVATION_BALANCE`. Additionally, validators with upgraded `0x02` credentials will no longer participate in the automatic withdrawal sweeps, making this tool is very valuable for Lighthouse validators.

Reviewer: ```suggestion
`EIP-7002` enables partial withdrawals from validators with `0x02` withdrawal credentials and balances exceeding the `MIN_ACTIVATION_BALANCE`. Additionally, validators with upgraded `0x02` credentials will no longer participate in the automatic withdrawal sweeps, making this tool very valuable for Lighthouse validators.
```

In order to request a partial withdrawal you must have access to the wallet set in the validator's withdrawal credentials and enough ETH to cover the withdrawal request and gas fees. Connect this wallet to the Siren dashboard to start withdrawing funds. All pending withdrawals will be visible in the same view for your convenience.

![partial-withdrawal](imgs/partial-withdrawal-siren.png)

### Partial Validator Top-ups

> If your validator's `EFFECTIVE_BALANCE` drops, or you've upgraded to `0x02` compounding withdrawal credentials, you can add additional funds. Simply connect any wallet to Siren and transfer the desired amount directly to your validators.

Reviewer 3: We might need to elaborate here. Specify how exactly to transfer money to validators.
